social-tw/social-tw-website
# [feature][relay] relay add stocketio for vote action

Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary

Implement socket io in relay server to emit vote events to clients.

## Linked Issue

close #171 

## Details

- Emit vote created/updated events with vote data
- Frontend subscribes and updates UI accordingly

## Impacted Areas

- packages/relay/src/routes/vote.ts - Emit events
- packages/relay/src/singletons/SocketManager.ts - Socket io server
- packages/relay/src/types/SocketTypes.ts - Events types
- packages/relay/test/* - Added tests

## Tests

- Added integration test using socket io client to validate events
- Simulates client side socket interactions

## Possible Impacts

- Real-time vote updates for better UX

## Verification Steps

1. Start relay server
2. Subscribe to vote events on frontend
3. Create vote through API
4. Observe real-time event on frontend

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [x] Implement socket io server
-   [x] Emit vote events

## Checklist
